Code/MinimalLib/tests/tests.js

```diff
@@ -214,16 +333,109 @@ M  END`;
 }
 
 
+function test_accept_failure() {
+    var template_molblock = `
+     RDKit          2D
+
+  9  9  0  0  0  0  0  0  0  0999 V2000
+   -0.8929    1.0942    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -2.1919    0.3442    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -2.1919   -1.1558    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.8929   -1.9059    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4060   -1.1558    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4060    0.3442    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -3.4910    1.0942    0.0000 R1  0  0  0  0  0  0  0  0  0  0  0  0
+    1.7051    1.0942    0.0000 R2  0  0  0  0  0  0  0  0  0  0  0  0
+   -3.4910   -1.9059    0.0000 R3  0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  2  0
+  2  3  1  0
+  3  4  2  0
+  4  5  1  0
+  5  6  2  0
+  6  1  1  0
+  6  8  1  0
+  3  9  1  0
+  2  7  1  0
+M  RGP  3   7   1   8   2   9   3
+M  END
+`;
+    var mol_molblock = `
+     RDKit          2D
+
+  9  9  0  0  0  0  0  0  0  0999 V2000
+   -0.8929    1.0942    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
+   -2.1919    0.3442    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -2.1919   -1.1558    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.8929   -1.9059    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4060   -1.1558    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4060    0.3442    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -3.4910    1.0942    0.0000 F   0  0  0  0  0  0  0  0  0  0  0  0
+    1.7051    1.0942    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
+   -3.4910   -1.9059    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  2  0
+  2  3  1  0
+  3  4  2  0
+  4  5  1  0
+  5  6  2  0
+  6  1  1  0
+  6  8  1  0
+  3  9  1  0
+  2  7  1  0
+M  END
+`;
+    var template_ref = RDKitModule.get_mol(template_molblock);
+    var mol = RDKitModule.get_mol(mol_molblock);
+    var hasThrown = false;
+    try {
+        mol.generate_aligned_coords(template_ref, false, true, false);
+    } catch (e) {
+        hasThrown = true;
+    }
+    assert(hasThrown);
+    assert.equal(mol.get_molblock(), mol_molblock);
+}
+
+function test_get_mol_no_kekulize() {
+    var mol = RDKitModule.get_mol("c");
+    assert(!mol.is_valid());
+    mol = RDKitModule.get_mol("c", JSON.stringify({kekulize: false}));
+    assert(mol.is_valid());
+}
+
+
 initRDKitModule().then(function(instance) {
+    var done = {
```

Comment:
apologies for my JS-illiteracy: why is this here?

Reply:
Reading SMILES from file is asynchronous, so I need to poll for completion to avoid printing "Test completed" before the test is actually completed.